### PR TITLE
feat: Trigger Star Alignment from UI

### PR DESCRIPTION
**Pull request by williamputraintan**
Screenshot 2023-08-15 at 17 16 21

Screenshot 2023-08-15 at 17 15 29

### CODE REVIEW

**Comment by williamputraintan** [comment]
[Portal Release - Ocicat: Implement Portal automation integration for oncoanalyser](https://trello.com/c/sZFv7FhZ/1488-portal-release-ocicat-implement-portal-automation-integration-for-oncoanalyser)

**Comment by victorskl** [comment]
Thanks Will; will review this, first thing tomorrow..!

**Comment by victorskl** [comment]
Reviewing now..

File: src/containers/subjects-launch/SubjectWTSStarAlignment/aws.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { Auth } from '@aws-amplify/auth';
+import { InvokeCommand, LambdaClient } from '@aws-sdk/client-lambda';
+import { REGION } from '../../../config';
+
+const STAR_ALIGNMENT_LAMBDA_NAME = 'star-align-nf-batch-job-submission';
+
+export type StarAlignmentPayload = {
+  portal_id?: string;
```
Comment (victorskl) [inline src/containers/subjects-launch/SubjectWTSStarAlignment/aws.ts:8]: Will; could you please change this to `portal_run_id`.

@scwatts Also notice that, if we could harmonise this naming at job submission Lambda side to be called `portal_run_id` as well, please.

Thanks both..!

File: src/containers/subjects-launch/SubjectWTSStarAlignment/index.tsx [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+
+  const [input, setInput] = useState<StarAlignmentPayload | null>(null);
+
+  // Find out the associated LibraryId
+  const {
+    isLoading: isLoadingSubjectData,
+    isError: isErrorSubjectData,
+    data: subjectData,
+  } = usePortalSubjectDataAPI(subjectId);
+  const listOfLibrary = subjectData?.lims.map((o) => o.library_id);
```
Comment (victorskl) [inline src/containers/subjects-launch/SubjectWTSStarAlignment/index.tsx:45]: Will; could you also filter `Type==WTS`, please. Star Alignment is only valid for WTS library type, yes.

File: src/pages/subjects/SubjectLaunchPad/index.tsx [hunk 3]
```diff
@@ -4,15 +4,17 @@ import { Card } from 'primereact/card';
...
+import SubjectRNASumLaunch from '../../../containers/subjects-launch/SubjectRNASumLaunch';
+import SubjectWGSTNLaunch from '../../../containers/subjects-launch/SubjectWGSTNLaunch';
+import SubjectLaunchStarAlign from '../../../containers/subjects-launch/SubjectWTSStarAlignment';
 
 export enum launchPadOptions {
   NONE = '',
   RNASUM = 'rna-sum',
   GPL = 'gpl',
   WGS_TN = 'wgs-tn',
+  STAR_ALIGN = 'star-align',
```
Comment (victorskl) [inline src/pages/subjects/SubjectLaunchPad/index.tsx:17]: Nicely done, alias..!

**Review by victorskl — COMMENTED** [review]
Nicely done, Will. Minor comment to attend.

File: src/containers/subjects-launch/SubjectWTSStarAlignment/aws.ts (hunk 1 above)
Comment (scwatts) [reply src/containers/subjects-launch/SubjectWTSStarAlignment/aws.ts:8]: No worries!

**Review by victorskl — APPROVED** [review]
Thanks for updates, Will. LGTM!

**Pull request merged**